woodwork/table_accessor.py

```diff
@@ -207,7 +225,7 @@ def _check_schema(dataframe, schema):
     if not isinstance(schema, Schema):
         raise TypeError('Provided schema must be a Woodwork.Schema object.')
     if not _is_valid_schema(dataframe, schema):
-        raise ValueError('Invalid typing information presented at init. Cannot set Woodwork on DataFrame.')
+        raise ValueError('Woodwork typing information is not valid for this DataFrame.')
```

Comment:
Yeah, I was thinking something along those lines so we could pass in a custom message depending on where the error happened.

Reply:
So if we want slightly different error messages based on when we check that the schema is valid
- before accessing the schema - `Cannot access logical_types; Woodwork typing information is invalid`
- when doing `df.ww.init(schema)` directly - `Invalid typing information presented at init`
- inside of the wrapper - `performing the replace operation has invalidated the typing information`

And each of those as a prefix to the specific message that should help with fixing the issue - something like `Column 'id' dtype 'string' does not match the listed Logical Type Categorical's dtype 'categorical'`

Reply:
I think for the cases where we warn, instead of error, we might want to mention that the typing information can be reinitialized by calling df.ww.init(). Not real clear how this fits exactly with setting the schema to None on invalidation, but these messages do provide additional detail that would help the user understand what is going on.